dubbo-common/src/main/java/org/apache/dubbo/common/metrics/collector/DefaultMetricsCollector.java

```diff
@@ -0,0 +1,181 @@
+            AtomicLong count = succeedRequests.computeIfAbsent(metric, k -> new AtomicLong(0L));
+            count.incrementAndGet();
+
+            publishEvent(new NewRequestEvent(metric, NewRequestEvent.Type.SUCCEED));
+        }
+    }
+
+    public void increaseFailedRequests(String interfaceName, String methodName, String parameterTypesDesc, String group, String version) {
+        if (isCollectEnabled()) {
+            MethodMetric metric = new MethodMetric(applicationName, interfaceName, methodName, parameterTypesDesc, group, version);
+            AtomicLong count = failedRequests.computeIfAbsent(metric, k -> new AtomicLong(0L));
+            count.incrementAndGet();
+
+            publishEvent(new NewRequestEvent(metric, NewRequestEvent.Type.FAILED));
+        }
+    }
+
+    public void increaseProcessingRequests(String interfaceName, String methodName, String parameterTypesDesc, String group, String version) {
+        if (isCollectEnabled()) {
+            MethodMetric metric = new MethodMetric(applicationName, interfaceName, methodName, parameterTypesDesc, group, version);
+            AtomicLong count = processingRequests.computeIfAbsent(metric, k -> new AtomicLong(0L));
+            count.incrementAndGet();
+        }
+    }
+
+    public void decreaseProcessingRequests(String interfaceName, String methodName, String parameterTypesDesc, String group, String version) {
+        if (isCollectEnabled()) {
+            MethodMetric metric = new MethodMetric(applicationName, interfaceName, methodName, parameterTypesDesc, group, version);
+            AtomicLong count = processingRequests.computeIfAbsent(metric, k -> new AtomicLong(0L));
+            count.decrementAndGet();
+        }
+    }
+
+    public void addRT(String interfaceName, String methodName, String parameterTypesDesc, String group, String version, Long responseTime) {
+        if (isCollectEnabled()) {
+            MethodMetric metric = new MethodMetric(applicationName, interfaceName, methodName, parameterTypesDesc, group, version);
+
+            AtomicLong last = lastRT.computeIfAbsent(metric, k -> new AtomicLong());
+            last.set(responseTime);
+
+            AtomicLong min = minRT.computeIfAbsent(metric, k -> new AtomicLong(Long.MAX_VALUE));
+            if (responseTime < min.longValue()) {
+                min.set(responseTime);
+            }
+
+            AtomicLong max = maxRT.computeIfAbsent(metric, k -> new AtomicLong(Long.MIN_VALUE));
+            if (responseTime > max.longValue()) {
+                max.set(responseTime);
+            }
+
+            AtomicLong total = totalRT.computeIfAbsent(metric, k -> new AtomicLong());
+            long newTotal = total.addAndGet(responseTime);
+
+            AtomicLong count = rtCount.computeIfAbsent(metric, k -> new AtomicLong());
+            AtomicLong avg = avgRT.computeIfAbsent(metric, k -> new AtomicLong());
+            long newAvg = newTotal / count.addAndGet(1);
+            avg.set(newAvg);
```

Comment:
The computing of avg value is unsafe when execute concurrently.  Lock it or compute avg value when fetching metrics.